pyperformance/data-files/benchmarks/bm_btree/run_benchmark.py

```diff
@@ -0,0 +1,474 @@
+                else:
+                    # Case 2c.
+                    extreme = upper_sibling.get_min_item()
+                    upper_sibling.delete(extreme[0])
+                    node.items = node.items + [extreme] + upper_sibling.items
+                    if not node.is_leaf():
+                        node.nodes = node.nodes + upper_sibling.nodes
+                    del self.items[p]
+                    del self.nodes[p + 1]
+            else:
+                if not is_big(node):
+                    if is_big(lower_sibling):
+                        # Case 3a1: Shift an item from lower_sibling.
+                        node.items.insert(0, self.items[p - 1])
+                        self.items[p - 1] = lower_sibling.items[-1]
+                        del lower_sibling.items[-1]
+                        if not node.is_leaf():
+                            node.nodes.insert(0, lower_sibling.nodes[-1])
+                            del lower_sibling.nodes[-1]
+                    elif is_big(upper_sibling):
+                        # Case 3a2: Shift an item from upper_sibling.
+                        node.items.append(self.items[p])
+                        self.items[p] = upper_sibling.items[0]
+                        del upper_sibling.items[0]
+                        if not node.is_leaf():
+                            node.nodes.append(upper_sibling.nodes[0])
+                            del upper_sibling.nodes[0]
+                    elif lower_sibling:
+                        # Case 3b1: Merge with lower_sibling
+                        node.items = (
+                            lower_sibling.items
+                            + [self.items[p - 1]]
+                            + node.items
+                        )
+                        if not node.is_leaf():
+                            node.nodes = lower_sibling.nodes + node.nodes
+                        del self.items[p - 1]
+                        del self.nodes[p - 1]
+                    else:
+                        # Case 3b2: Merge with upper_sibling
+                        node.items = (
+                            node.items + [self.items[p]] + upper_sibling.items
+                        )
+                        if not node.is_leaf():
+                            node.nodes = node.nodes + upper_sibling.nodes
+                        del self.items[p]
+                        del self.nodes[p + 1]
+                assert is_big(node)
+                node.delete(key)
+            if not self.items:
+                # This can happen when self is the root node.
+                self.items = self.nodes[0].items
+                self.nodes = self.nodes[0].nodes
+
+
+class BTree(collections.abc.MutableMapping):
+    """
+    Instance attributes:
+      root: BNode
+    """
+
+    __slots__ = ['root']
+
+    def __init__(self, node_constructor=BNode):
+        assert issubclass(node_constructor, BNode)
+        self.root = node_constructor()
```

Comment:
Is changing `node_constructor` ever used?

```suggestion
    def __init__(self):
        self.root = BNode()
```

Reply:
Done.